### PR TITLE
Tigger v1.6.0

### PR DESCRIPTION
**Pull request by razman786**
* Python 3 support.
* Ported from PyQt4 and Qwt 5, to PyQt 5 and Qwt 6.
* Depends on Tigger-LSM v1.7.0.
* Supports Ubuntu LTS versions 18.04, 20.04 and 21.04.
* Added installation script that automatically installs Tigger and dependencies.
* New dark mode theme.
* Anti-aliasing rendering.
* Support for High DPI displays and scaling.
* New customisable GUI interface with dockable, tabbed and floating windows.
* Exported PNG's now calculate maximum image resolution based on free memory resources.
* Added desktop GUI application launcher.
* Coordinate information is now sent to the terminal and the clipboard.
* Live Profile window is now activated upon click.
* Various fixes and refactoring.

### CODE REVIEW

**Comment by bennahugo** [comment]
Thanks @razman786 I will review this tomorrow. I think the test is a left over from the days prior to tigger-lsm being a package, but I will fix it as part of the review.

**Comment by razman786** [comment]
@bennahugo - some known issues, hopefully saves some time:

nown issues:

* Upon startup, a Qt warning is given on the terminal about the style setting, which is already set to fusion - GTK3/unity and Qt sometimes have this issue.
* On Ubuntu 21.04 upon startup, a Qt wayland warning is given on the terminal - new XServer backend in this new version of Ubuntu.
* Tigger application launcher is only available after logging on and off again - this seems to be most noticeable on a 'scorched earth' type VM test and may not be the case on a normal users machine.
* If using a small screen with a high scale setting in Ubuntu and Tigger in a window (not maximised full screen), for example 13 inch screen with 200% scale setting, the Tigger window may grow larger than the height of the screen. Lowering the scale to 150% or using Tigger in full screen, normally solves this issue - requires further user feedback, as the developer only has 2 screens (a very small one and a very large one).
* The 'Limit 4K' image export option is only shown in Linux distro's that do not override Qt's file dialog with native versions - Ubuntu is seemingly overriding them by default.
* Right-click on a model table shows only a basic menu without options - this has been left as an enhancement.
* The installer script installs astro-tigger-lsm v1.7.0 from source on Ubuntu 18.04, as the pip method has issues.

File: TigGUI/Tools/source_selector.py [hunk 1]
```diff
@@ -137,27 +142,35 @@ def _reset_percentile(self):
...
                 if hasattr(src, tag):
-                    value = float(getattr(src, tag))
-                else:
+                    # test if item can be cast to float
+                    try:
+                        float(getattr(src, tag))
+                    except:
+                        continue
+                    else:
+                        value = float(getattr(src, tag))
```
Comment (bennahugo) [inline TigGUI/Tools/source_selector.py:162]: Note to self check if this supports listing +dE string tags???

File: setup.py [hunk 2]
```diff
@@ -17,22 +27,21 @@
     ]
 }
 
-extras_require = {
-    'venv': ['vext.pyqt4'],
-}
-
```
Comment (bennahugo) [inline setup.py:23]: I think this is still needed (at least for me in my virtualenv - vext.pyqt5

**Review by bennahugo — APPROVED** [review]
This looks fantastic. Minor niggles. Will fix the docker installation and add an action to test and deploy. Just need to test those tag filtering - not all tags are float, some can be boolean and just be named for instance

**Comment by bennahugo** [comment]
ok to test

**Comment by bennahugo** [comment]
Ok I'm now happy with this. There is an integration test that fires tigger up in XVFB and loads in a fits file and pybdsf gaul catalog @o-smirnov. The output is verified for errors (stdout and stderr is piped into a file and verified. This ofc. does not guarantee that the gui is 100% correct but it does at least give some sort of validation in integration of all the modules.

Furthermore I've added "==" selection so you can select on "dE" tags for instance - something that has long since not worked like it should. Where sources do not have a tag it is assumed their default is -inf. The min and maxes are left at the value set for the custom tag on the remainder. I also added an option to unselect - this has been sorely missing.

**Comment by bennahugo** [comment]
Only you have the foo to actually merge this. Can you please allow "ratt-priv-ci" access to this repo so it can set statuses?

**Comment by bennahugo** [comment]
I'm happy this passes:
For commit 7391a5e2e7e83f1 (above) at https://jenkins.meqtrees.net/job/PR_tigger/3/
```
*******************************
 Integration Test 
*******************************

Starting X11 Virtual Frame Buffer and waiting for 10s for Tigger to load and run


Welcome to Tigger 1.6.0
Please wait a second while the GUI starts up.
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'
Detected screen size 1024x768, usable area 1024x768
Loaded FITS image /tmp/tigger/test-files/star_model_image.fits
Loaded 17 sources from 'Gaul' file /tmp/tigger/test-files/cat.gaul


Output looks ok as far as we can tell


*******************************
 Integration Test Passed 
*******************************
Removing intermediate container 0b4f6d2c21f3
 ---> 1403ade83ead
Step 8/8 : ENTRYPOINT .local/bin/tigger
 ---> Running in 4ed6d33434df
Removing intermediate container 4ed6d33434df
 ---> d671955e9d89
Successfully built d671955e9d89
Successfully tagged tigger:3
+ status=0
+ [ 0 -eq 0 ]
+ echo Jenkinsfile.sh build job ran successfully
Jenkinsfile.sh build job ran successfully
+ exit 0
```

**Comment by razman786** [comment]
@bennahugo - ratt-priv-ci added to repo

**Comment by bennahugo** [comment]
retest this please

**Comment by bennahugo** [comment]
Alright it works - @o-smirnov I think you can merge her

**Pull request merged**